client/post-editor/post-editor.jsx

```diff
@@ -854,18 +857,11 @@ const PostEditor = React.createClass( {
 	},
 
 	getEditorMode: function() {
-		var editorMode = 'tinymce'
-		if ( this.props.preferences ) {
-			if ( this.props.preferences[ 'editor-mode' ] ) {
-				editorMode = this.props.preferences[ 'editor-mode' ];
-			}
-
-			if ( ! this.recordedDefaultEditorMode ) {
-				analytics.mc.bumpStat( 'calypso_default_editor_mode', editorMode );
-				this.recordedDefaultEditorMode = true;
-			}
+		const editorMode = this.props.editorModePreference || 'tinymce';
+		if ( ! this.recordedDefaultEditorMode ) {
```

Comment:
This is not quite the same effect as existed previously, as before we had guarded against recording the default mode until after preferences have loaded. With this refactor, we can't guarantee that the recorded mode is in-fact the user preference, as it may simply be the fallback value while preferences are loading.